### DISCUSSION TITLE
Issues with the 2.0 Colab on Free/Cheap Version

### DISCUSSION THREAD

**Discussion by MasterDenis** [TheLastBen/fast-stable-diffusion#799] [General]
Hey @TheLastBen and everyone!
So lately I had some rough luck with this colab and the low vram counterpart, this is the latest exampe:
![image](https://user-images.githubusercontent.com/61897713/205110463-d469e277-8752-4e7e-ae15-d6aa2bf1c182.png)
I could have it working yesterday though however when rendering/generating the image would not appear on the gradio interface for automatic1111.
That is one issue, the other issue would be more of a feedback, if we could have somehow the low-vram version to use the main SD folder in the google drive directory (unless it is incompatible) and lastly if there is a possibility to add that low-vram version/colab on the main one because checking, waiting and then having the link show up may feel unclean.
Thanks for the efforts so far.

**Comment by ivangalucho** [comment]
Hi! i think everything its broken due to colab updating python automatic1111 for sd v1.5 also not working
![204965753-df057ad9-4db1-4f30-b133-7de4a797c9c0](https://user-images.githubusercontent.com/107110452/205131526-3fe37bda-0250-485d-b2c0-3a9799a32bb7.png)

**Reply by MasterDenis** [reply]
> Hi! i think everything its broken due to colab updating python automatic1111 for sd v1.5 also not working ![204965753-df057ad9-4db1-4f30-b133-7de4a797c9c0](https://user-images.githubusercontent.com/107110452/205131526-3fe37bda-0250-485d-b2c0-3a9799a32bb7.png)

that is a different issue on the main colab but yeah I saw it too now.

**Comment by TheLastBen** [comment]
@altahookah that issue has been already fixed, use the latest colab

**Reply by ivangalucho** [reply]
Hi! ben! thanks for the answer! im still getting
`Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 14, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization, ui_tempdir
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 31, in <module>
    ldm.modules.attention.BasicTransformerBlock.ATTENTION_MODES["softmax-xformers"] = ldm.modules.attention.CrossAttention
AttributeError: type object 'BasicTransformerBlock' has no attribute 'ATTENTION_MODES'`

what do you mean by the "latest" colab? im using the one on the main github of yours
[https://github.com/TheLastBen/fast-stable-diffusion](url)

[https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast_stable_diffusion_AUTOMATIC1111.ipynb](url)

![image](https://user-images.githubusercontent.com/107110452/205219902-e6c3bb2d-cc65-4dd9-afd7-a4620aaafc03.png)

thanks again!!

**Reply by TheLastBen** [reply]
remove the folder sd from your gdrive, your folder structure is old and has changed since V2

**Reply by ivangalucho** [reply]
thaaaaaanks!!!! this worked for me!! ill support you , very very kind and amazing work you are doing!!!

**Comment by TheLastBen** [comment]
@MasterDenis after they updated to python 3.8, the colab VM seems to take a bit more RAM and causes sd to error out, I'm trying to find a solution. I seperated the two colabs because connecting to gdrive consumes RAM and also causes RAM error. the separate colab is only temporary

**Reply by redsparrowgit** [reply]
The answer is one, google blocked so that the people (street) would not have access to this technology so easily. The only thing left is to work offline. Restoring the old version of python

**Reply by TheLastBen** [reply]
yes, clearly they did it on purpose to force people to buy pro

**Reply by TheLastBen** [reply]
yes, use only regular pro, no need for the pro +

**Reply by TheLastBen** [reply]
Runtime-> Change runtime type -> Runtime shape -> High-RAM
keep gpu to standard

**Reply by redsparrowgit** [reply]
Thank You for info.

**Reply by MasterDenis** [reply]
I see, Indeed that makes sense, mounting on the drive gave a few good options such as having a static folder that stores settings/embendings etc so that would be a good priority until a solution is found for longterm, but yeah take your time. thanks.

**Reply by MasterDenis** [reply]
Also @TheLastBen this error still happens for a couple of days now on the alternative colab.
![image](https://user-images.githubusercontent.com/61897713/205303753-5bdbf05d-e6bb-4d30-afb5-5327beecc11f.png)

**Reply by MasterDenis** [reply]
@TheLastBen I noticed the latest notebook, assuming it works with free/cheap now, I was wondering about this error:

![image](https://user-images.githubusercontent.com/61897713/205652204-700a1d35-91ed-4c0b-b5bc-1867223682dc.png)

I will try to delete the folder completely and such to let it freshly install again, I will keep you updated.

Thanks

**Reply by TheLastBen** [reply]
you didn't select the right version of the model

**Reply by MasterDenis** [reply]
> you didn't select the right version of the model

I did, the issue was that I had to completely delete the folder, even the initial folder "sd", now it works.

**Comment by Lastofthefirst** [comment]
v2 on free collab seems to be working once I deleted my drive files with the 512 settings. Thanks for the good work. ❤️

**Reply by MasterDenis** [reply]
> v2 on free collab seems to be working once I deleted my drive files with the 512 settings. Thanks for the good work. ❤️

Same, something seemed to have remained from the prior installation.

**Comment by THX1139b** [comment]
For me I get the same exact error as the OP. Deleting the SD folder has worked in the past but not now. I always use the latest link for the notebook. I've tried it a dozen times today and yesterday but unfortunately the web UI is never loaded. I love this stuff so it will be deprerssing if it will never work for me again.

**Reply by TheLastBen** [reply]
everything is fixed since a few hours ago, can you paste the error log here ?

**Reply by THX1139b** [reply]
Thank you for answering. I'm afraid I'm unable to find a log file. Don't know what it is called or where it is supposed to be. Looked in all the SD sub folders and I only see HEAD logs. In the notebook it is identical to the OP's screenshot posted here.

I think my problem is that the RAM maxes out. It goes all the way to 16 point something and when it gets full it all stops. I do have Colab Pro but perhaps this is no longer enough for the V2-768 work.

**Reply by THX1139b** [reply]
<img width="791" alt="Screenshot 2022-12-08 at 14 02 36" src="https://user-images.githubusercontent.com/111381350/206453481-15af1c64-ebb5-499c-8587-c957fc91309e.png">
Well, nearly identical.

**Reply by TheLastBen** [reply]
is the roaldamu5.ckpt in fp16 or fp32 ?

**Reply by THX1139b** [reply]
It's fp32.

**Reply by TheLastBen** [reply]
use fp16

**Reply by THX1139b** [reply]
Fair enough. I'll give it a go, thank you!

EDIT: yes! I can confirm that it works. RAM use on Colab went almost all the way up to max but it the web-UI on first attempt. Thank you again for your great work and help!

**Reply by redsparrowgit** [reply]
> Fair enough. I'll give it a go, thank you!
> 
> EDIT: yes! I can confirm that it works. RAM use on Colab went almost all the way up to max but it the web-UI on first attempt. Thank you again for your great work and help!

That's it in summary - what do i need to do to get it working ?
